ga4gh/datamodel/reads.py

```diff
@@ -91,6 +91,121 @@ def setFlag(flagAttr, flag):
+        Returns an iterator over the specified reads
+        """
+        # TODO If reference is None, return against all references,
+        # including unmapped reads.
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        # TODO deal with errors from htslib
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        for readAlignment in readAlignments:
+            tags = dict(readAlignment.tags)
+            if readGroup is None:
+                alignmentReadGroup = None
+                if 'RG' in tags:
+                    alignmentReadGroupId = tags['RG']
+                    alignmentReadGroup = self._readGroupLocalIdMap.get(
+                        alignmentReadGroupId, None)
+                yield self.convertReadAlignment(
+                    readAlignment, readGroupSet, alignmentReadGroup)
+            else:
+                if self._filterReads:
+                    if 'RG' in tags and tags['RG'] == self._localId:
+                        yield self.convertReadAlignment(
+                            readAlignment, readGroupSet, readGroup)
+                else:
+                    yield self.convertReadAlignment(
+                        readAlignment, readGroupSet, readGroup)
+
+    def convertReadAlignment(self, read, readGroupSet, readGroup):
+        """
+        Convert a pysam ReadAlignment to a GA4GH ReadAlignment
+        """
+        samFile = self.getFileHandle(self._samFilePath)
+        # TODO fill out remaining fields
+        # TODO refine in tandem with code in converters module
+        ret = protocol.ReadAlignment()
+        ret.fragmentId = 'TODO'
+        if read.query_qualities is None:
+            ret.alignedQuality = []
+        else:
+            ret.alignedQuality = list(read.query_qualities)
+        ret.alignedSequence = read.query_sequence
+        if SamFlags.isFlagSet(read.flag, SamFlags.READ_UNMAPPED):
+            ret.alignment = None
+        else:
+            ret.alignment = protocol.LinearAlignment()
+            ret.alignment.mappingQuality = read.mapping_quality
+            ret.alignment.position = protocol.Position()
+            ret.alignment.position.referenceName = samFile.getrname(
+                read.reference_id)
+            ret.alignment.position.position = read.reference_start
+            ret.alignment.position.strand = protocol.Strand.POS_STRAND
+            if SamFlags.isFlagSet(read.flag, SamFlags.READ_REVERSE_STRAND):
+                ret.alignment.position.strand = protocol.Strand.NEG_STRAND
+            ret.alignment.cigar = []
+            for operation, length in read.cigar:
+                gaCigarUnit = protocol.CigarUnit()
+                gaCigarUnit.operation = SamCigar.int2ga(operation)
+                gaCigarUnit.operationLength = length
+                gaCigarUnit.referenceSequence = None  # TODO fix this!
```

Comment:
Would you please make an issue to go with this TODO? Does it require a lookup on the reference to perform properly?